docs/architecture/adr-027-deterministic-protobuf-serialization.md

```diff
@@ -8,7 +8,17 @@
 
 Proposed
 
-## Context
+## Abstract
+
+We need fully deterministic Protobuf serializaiton which works across many languages / clients. Tthe major use-case is to sign a structure, and this is the flow:
+
+1. Serialize structure -> raw bytes
+1. Hash raw bytes.
```

Comment:
For signing tx, we actually don't hash the bytes.

Reply:
Are you sure? Probably this is on the tendermint side.

Reply:
The hash is done in the signing library, I will make it clear in the description here.